docs/guide_cog_creation.rst

```diff
@@ -162,3 +162,43 @@ Additional resources
 Be sure to check out the :doc:`/guide_migration` for some resources
 on developing cogs for V3. This will also cover differences between V2 and V3 for
 those who developed cogs for V2.
+
+
+---------------------------
+Guidelines for Cog Creators
+---------------------------
+
+The following are a list of guidelines cog creators should strive to follow.
+Not all of these are strict requirements (some are) but are all generally advisable.
+
+1. Cog names should be TitleCased and only contain alphabetic characters.
+2. Cogs and commands should have docstrings suitable for use in help output.
+  - This one is slightly flexible if using other methods of setting help.
+3. Don't prevent normal operation of the bot without the user opting into this.
+  - This includes as a side effect by blocking the event loop.
+4. If your cog uses logging:
+  - The namespace for logging should be: ``red.your_repo_name.cog_name``.
+  - Print statements are not a substitute for proper logging.
+5. If you use asyncio.create_task, your tasks need to:
+  - Be cancelled on cog unload.
+  - Handle errors.
+6. Event listeners should exit early if it is an event you don't need.
```

Comment:
This is a bit unclear, perhaps something more like "Event listeners should have checks allowing them to exit early for events that are not needed". Even that isn't as good as I'd like it and I don't know how to make it clear without an example.